src/datasets/utils/file_utils.py

```diff
@@ -499,7 +501,8 @@ def get_from_cache(
         if scheme == "ftp":
             connected = ftp_head(url)
         elif scheme not in ("http", "https"):
-            response = fsspec_head(url)
+            response = fsspec_head(url, storage_options=storage_options)
+            # use the hash of the response as a pseudo ETag to detect changes
```

Comment:
```suggestion
```

Reply:
Thanks, this fell through the cracks of a rebase!